pkg/docker/parseimage_test.go

```diff
@@ -76,5 +98,15 @@ func TestParseImageIDString(t *testing.T) {
 		if sha != tc.sha && tc.shouldPass {
 			t.Errorf("[%s] sha is wrong.  Expected %s got %s", tc.description, tc.sha, sha)
 		}
+		if !tc.shouldPass && err == nil {
+			t.Errorf("The error should not be empty, description: %s , prefix: %s, name: %s, sha: %s", tc.description, tc.prefix, name, sha)
+		}
+	}
+
+	imageID := "docker://sha256:cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043"
```

Comment:
We should be able to fold this test into the other iterative tests by re-working things a little.  Move the static @sha256 bit into the sha field for each test then should be able to put this test as a usual test case.